Code/LauncherUnified/CMakeLists.txt

```diff
@@ -62,6 +82,21 @@ if(PAL_TRAIT_BUILD_SERVER_SUPPORTED)
                 Legacy::CryCommon
     )
 
+    ly_add_target(
+        NAME Launcher.Headless.Server.Static STATIC
+        NAMESPACE AZ
+        FILES_CMAKE
+            launcher_server_files.cmake
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                .
+        BUILD_DEPENDENCIES
+            PRIVATE
+                AZ::AzCore
+                AZ::AzGameFramework.Headless
+                Legacy::CryCommon
+    )
+
```

Comment:
This CMake target isn't needed.
Only the `Launcher.Headless.Static` target is needed.

It looks like you added this because the `Launcher.Server.Static` target had a dependency on AZ:::GameFramework.

However the only file in the in the Launcher.Server.Static target is [Server.cpp](https://github.com/o3de/o3de/blob/development/Code/LauncherUnified/Server.cpp) and that has **no dependencies at all**.

Instead what I recommend doing is to remove the PRIVATE build dependencies from the Launcher.Server.Static target and to remove this TARGET as well

Reply:
Actually the `Launcher.Headless.Server.Static` target would be need if the `O3DE_HEADLESS_SERVER` define needs to be checked.

It is currently not needed in this revision of the PR because the Server.cpp is **missing** `IsHeadless` implementation

Reply:
This was actually done to differentiate the dependencies to `AZ::AzGameFramework` and `AZ::AzGameFramework.Headless`. However, I can still remove `Launcher.Headless.Server.Static` and mark the dependency to AZ::AzGameFramework in the `launcher_generator.cmake`